#! /bin/sh /usr/share/dpatch/dpatch-run
## 03-location-of-help-files.dpatch by Tiago Saboga <tiagosaboga@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Change location of help files.

@DPATCH@
diff -urNad omegat-1.7.2.dfsg0~/src/org/omegat/gui/HelpFrame.java omegat-1.7.2.dfsg0/src/org/omegat/gui/HelpFrame.java
--- omegat-1.7.2.dfsg0~/src/org/omegat/gui/HelpFrame.java	2007-09-22 14:01:02.000000000 -0300
+++ omegat-1.7.2.dfsg0/src/org/omegat/gui/HelpFrame.java	2007-11-10 12:43:20.000000000 -0200
@@ -210,8 +210,7 @@
         StringBuffer templateText = new StringBuffer(1024);
         try {
             BufferedReader templateFile = new BufferedReader(new FileReader(
-                  StaticUtils.installDir() + File.separator
-                + OConsts.HELP_DIR + File.separator
+                  OConsts.HELP_DIR + File.separator
                 + OConsts.HELP_LANG_INDEX));
             for (String line = templateFile.readLine();
                  line != null;
@@ -229,7 +228,7 @@
         StringBuffer translations = new StringBuffer(1024);
         translations.append("<table>\n");
         File docDir = new File(
-            StaticUtils.installDir() + File.separator + OConsts.HELP_DIR);
+            OConsts.HELP_DIR);
         File[] subDirs = docDir.listFiles();
         Arrays.sort(subDirs); // sort on alphabetical order
         for (int i = 0; i < subDirs.length; i++) {
@@ -430,8 +429,7 @@
     {
         try {
             return   "file:"                                                          // NOI18N
-                   + (new File(  StaticUtils.installDir()
-                              + File.separator + OConsts.HELP_DIR + File.separator
+                   + (new File(  OConsts.HELP_DIR + File.separator
                               + m_language + File.separator + file)).getCanonicalPath();
         }
         catch (IOException exception) {
@@ -476,8 +474,7 @@
     private static String getDocVersion(String locale) {
         // Check if there's a manual for the specified locale
         // (Assume yes if the index file is there)
-        File index = new File(StaticUtils.installDir()
-            + File.separator + OConsts.HELP_DIR
+        File index = new File(OConsts.HELP_DIR
             + File.separator + locale
             + File.separator + OConsts.HELP_HOME);
         if (!index.exists())
@@ -485,8 +482,7 @@
 
         // Check if the doc dir for the specified locale
         // contains a file containing the doc version
-        File v = new File(StaticUtils.installDir()
-            + File.separator + OConsts.HELP_DIR
+        File v = new File(OConsts.HELP_DIR
             + File.separator + locale
             + File.separator + "version.properties");
         if (!v.exists())
diff -urNad omegat-1.7.2.dfsg0~/src/org/omegat/gui/main/MainWindow.java omegat-1.7.2.dfsg0/src/org/omegat/gui/main/MainWindow.java
--- omegat-1.7.2.dfsg0~/src/org/omegat/gui/main/MainWindow.java	2007-09-20 09:52:52.000000000 -0300
+++ omegat-1.7.2.dfsg0/src/org/omegat/gui/main/MainWindow.java	2007-11-10 12:43:20.000000000 -0200
@@ -66,6 +66,7 @@
 import javax.swing.text.BadLocationException;
 import javax.swing.undo.CannotRedoException;
 import javax.swing.undo.CannotUndoException;
+import net.roydesign.mac.MRJAdapter;
 
 import org.omegat.core.ProjectProperties;
 import org.omegat.core.StringEntry;
@@ -502,8 +503,7 @@
         {
             String language = detectInstantStartLanguage();
             String filepath =
-                    StaticUtils.installDir()
-                    + File.separator + OConsts.HELP_DIR + File.separator
+                    OConsts.HELP_DIR + File.separator
                     + language + File.separator
                     + OConsts.HELP_INSTANT_START;
             JTextPane instantArticlePane = new JTextPane();
@@ -534,16 +534,14 @@
         String country  = java.util.Locale.getDefault().getCountry().toUpperCase();
 
         // Check if there's a translation for the full locale (lang + country)
-        File isg = new File(StaticUtils.installDir()
-            + File.separator + OConsts.HELP_DIR
+        File isg = new File(OConsts.HELP_DIR
             + File.separator + language + "_" + country
             + File.separator + OConsts.HELP_INSTANT_START);
         if (isg.exists())
             return language + "_" + country;
 
         // Check if there's a translation for the language only
-        isg = new File(StaticUtils.installDir()
-            + File.separator + OConsts.HELP_DIR
+        isg = new File(OConsts.HELP_DIR
             + File.separator + language
             + File.separator + OConsts.HELP_INSTANT_START);
         if(isg.exists())
diff -urNad omegat-1.7.2.dfsg0~/src/org/omegat/util/OConsts.java omegat-1.7.2.dfsg0/src/org/omegat/util/OConsts.java
--- omegat-1.7.2.dfsg0~/src/org/omegat/util/OConsts.java	2006-12-02 01:02:34.000000000 -0200
+++ omegat-1.7.2.dfsg0/src/org/omegat/util/OConsts.java	2007-11-10 12:43:20.000000000 -0200
@@ -56,7 +56,7 @@
     // help
     public static final String HELP_HOME          = "index.html";            // NOI18N
     public static final String HELP_INSTANT_START = "instantStartGuideNoTOC.html";// NOI18N
-    public static final String HELP_DIR           = "docs";                   // NOI18N
+    public static final String HELP_DIR           = "/usr/share/omegat/html-template";                   // NOI18N
     public static final String HELP_LANG_INDEX    = "languageIndex.html";                  // NOI18N
     
     // status file consts
